File: app/src/main/java/edu/gatech/spellastory/domain/Category.java
package edu.gatech.spellastory.domain;

public class Category {

    private int code;
    private String name;

    public Category(int code, String name) {
        this.code = code;
        this.name = name;
    }

    public int getCode() {
        return code;
    }

    public String getName() {
        return name;
    }

}

File: app/src/main/java/edu/gatech/spellastory/domain/Category.kt
package edu.gatech.spellastory.domain

import java.io.Serializable

class Category(val code: Int, val name: String) : Serializable {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Category

        if (code != other.code) return false

        return true
    }

    override fun hashCode() = code

    override fun toString() = "Category(name='$name')"
}